setdate.js

```diff
@@ -0,0 +1,82 @@
+
+var myDate = (function(){
+
+  var __date = null;
+
+  function __checkIfInstantiated(){
+    if (__date) { return __date; } else { return new Date(); };
+  };
+
+  function __milliseconds(){
+    var date = __checkIfInstantiated();
+
+    return date.getTime();
+  };
+
+  function __formatted(){
+    var date = __checkIfInstantiated();
+    var month = getMonthName();
+    var day = date.getDate();
+    var year = date.getFullYear();
+
+    return month + " " + day + ", " + year;
+
+  };
+
+  function setDate(value){
+    value = value || Date.now();
+    __date = new Date(value);
+  };
+
+  function getDate(options){
+    options = options || { format: "milliseconds" };
+    var format = options.format;
+
+    return window["__" + format]();
+  };
+
+  function getDayName(){
+    var date = __checkIfInstantiated();
+    var days = ["Sunday", "Monday", "Tuesday", "Wednesday",
```

Comment:
Using an array is a good way to implement this method, but rather than declaring your array here, it would be preferable to declare it inside your module, but outside of this method. Every time this method is called it will re-declare and reserve memory for this array. By moving it out of this method but still putting it inside your module you'll gain a little efficiency.